Sources/XCTest/Public/Asynchronous/XCTWaiter.swift

```diff
@@ -356,22 +349,15 @@ open class XCTWaiter {
 
 private extension XCTWaiter {
     func primitiveWait(using runLoop: RunLoop, duration timeout: TimeInterval) {
-        // The contract for `primitiveWait(for:)` explicitly allows waiting for a shorter period than requested
-        // by the `timeout` argument. Only run for a short time in case `cancelPrimitiveWait()` was called and
-        // issued `CFRunLoopStop` just before we reach this point.
-        let timeIntervalToRun = min(0.1, timeout)
-
-        // RunLoop.run(mode:before:) should have @discardableResult <rdar://problem/45371901>
-        _ = runLoop.run(mode: .default, before: Date(timeIntervalSinceNow: timeIntervalToRun))
+        
+        self.runLoopSource = RunLoop._Source()
+        self.runLoop?._add(self.runLoopSource!, forMode: .default)
+        _ = self.runLoop?.run(until: .init(timeIntervalSinceNow: timeout))
     }
 
     func cancelPrimitiveWait() {
-        guard let runLoop = runLoop else { return }
-#if os(Windows)
-        runLoop._stop()
-#else
-        CFRunLoopStop(runLoop.getCFRunLoop())
-#endif
+        
```

Comment:
Similar

Reply:
Added the whitespace at the start and end of a method for readability. Let me know if that's not a recommended way, I can remove it.

Reply:
Added a newline before starting the method for readability. Let me know if it's not a recommended way, i can remove it.

Reply:
@compnerd  have removed the whitespace